src/lib/kernel/Linux.cc

```diff
@@ -524,20 +532,28 @@ int64_t Linux::readlinkat(int64_t dirfd, const std::string& pathname, char* buf,
   if (pathname == "/proc/self/exe") {
     // Resolve absolute path
     char absolutePath[LINUX_PATH_MAX];
-    realpath(processState.path.c_str(), absolutePath);
+    char* output = realpath(processState.path.c_str(), absolutePath);
+    if (output == nullptr) {
+      // Something went wrong
+      std::cerr << "[SimEng:readlinkat] realpath failed with errno = " << errno
+                << std::endl;
+      exit(EXIT_FAILURE);
+    }
```

Comment:
There are comments in another file about this logic, adding comment to ensure we don't miss this instance if the other logic is changed

Reply:
Will see what the solution is and apply here

Reply:
Updated